src/main/java/seedu/duke/data/timetable/Reminder.java

```diff
@@ -0,0 +1,60 @@
+package seedu.duke.data.timetable;
+
+import java.time.LocalDate;
+
+public class Reminder implements Comparable<Reminder> {
+    private Event event;
+    private LocalDate dateToRemind;
+
+    /**
+     * Constructor for a Reminder. Includes what event to remind about as well as what date to remind on.
+     *
+     * @param event Event to remind
+     * @param dateToRemind When to remind
+     */
+    public Reminder(Event event, LocalDate dateToRemind) {
+        this.event = event;
+        this.dateToRemind = dateToRemind;
+    }
+
+    public LocalDate getDateToRemind() {
+        return dateToRemind;
+    }
+
+    /**
+     * Returns whether this reminder should be raised to the user.
+     * If the date provided is equal the dateToRemind stored in the reminder, it returns true.
+     *
+     * @param date Date to check
+     * @return Whether the reminder should be raised.
+     */
+    public boolean toRemind(LocalDate date) {
+        return dateToRemind.equals(date);
+    }
+
+    @Override
+    public String toString() {
+        return event.toReminderString();
+    }
+
+    @Override
+    public int compareTo(Reminder o) {
+        return dateToRemind.compareTo(o.dateToRemind);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
The parameter has to be an Object? Is there any instance that this will be used to compare non-Reminder objects?

Reply:
I used Object cause it was more type-safe. I was considering that when the timetable expands and has multi-year usage, we may have issues with recurring events and their reminders so I was implementing the equals function to be able to compare reminders. However, I revamped how the recurring events are displayed so this function actually has no more use.